packages/plugin-commands-installation/src/installDeps.ts

```diff
@@ -199,34 +199,42 @@ when running add/update with the --workspace option')
     }
   }
 
-  const updateMatch = opts.update && (params.length > 0) ? createMatcher(params) : null
+  let currentInput = [...params]
+
+  if (opts.update) {
+    const ignoredPackages = (manifest.pnpm?.updateConfig?.ignoreDependencies ?? [])
+    currentInput = [...ignoredPackages.map(pkg => `!${pkg}`), ...currentInput]
```

Comment:
I thought we already discussed that the ignore list will only be active when pnpm update is called without args. So why do you add currentInput?

Reply:
I thought it would be ok to append the ignored packages to the beginning of currentInput with `!` at the beginning of currentInput because the successor params will override `!` packages. 

However, I have rethought that it is not friendly to ignore packages even if they are explicitly specified `*` params, so I will append only if the params are empty 🙏 .

Reply:
Fixed in [dfc8b5e](https://github.com/pnpm/pnpm/pull/5408/commits/dfc8b5e412f29b8c6e90e3c184a4e9a1b9d86948)